### ISSUE TITLE
tf keras upsample2d operator produces wrong results on tflite conversion

### ISSUE THREAD

**Issue by anilsathyan7** [tensorflow/tensorflow#45090]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- TensorFlow installed from (source or binary): source and binary
- TensorFlow version (or github SHA if from source): 2.3.0, tf-nightly, 2.5.0(latest source)

**Command used to run the converter or code if you’re using the Python API**
If possible, please share a link to Colab/Jupyter/any notebook.
Colab Link: https://colab.research.google.com/drive/1QZo8QZubmiiXIxc0O8SZEfEsfaxN5kxR?usp=sharing

```
converter = tf.lite.TFLiteConverter.from_keras_model(model)
tflite_model = converter.convert()
open("mnv3_test.tflite", "wb").write(tflite_model)
```

**Failure details**
The conversion is successful, but the generated model is wrong,

The tf keras model with **upsample2d layers(bilinear interpolation)** correctly produces a h5 file; but the converted tflite model has **extra layers** and there is **shape mismatch**.The **output shape becomes 1x1** instead of original upsampled size. The error occurs in tf 2.3.0, tf nightly and latesr source (dev 2.5.0). The conversion produces **additonal shape, mul and strided slice** layer(not supported by hardware accelerators).

The problem persists even if we use  **ResizeNearestNeighbo**r interpolation.

**Minimal Example**

```
import tensorflow as tf
from tensorflow.keras.layers import Input, Conv2D, UpSampling2D
from tensorflow.keras.models import Model
from tensorflow.keras.optimizers import Adam

def get_model():
    
    ip = Input(shape=(256,256,3))
    x= Conv2D(kernel_size=3, filters=32, padding ='same')(ip)
    x = UpSampling2D( size=(2, 2), interpolation='bilinear')(x)
    x= Conv2D(kernel_size=3, filters=32, padding ='same')(x)
    
    
    model = Model(inputs=ip, outputs=x)
    
    model.compile(optimizer=Adam(lr=1e-4),
              loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
              metrics=['accuracy']) # Ensure you have sparse labels
    return model

model=get_model()
converter = tf.lite.TFLiteConverter.from_keras_model(model)
tflite_model = converter.convert()
open("bilinear.tflite", "wb").write(tflite_model)
```

**Models**
[mnv3_resize_bilinear_test.zip](https://github.com/tensorflow/tensorflow/files/5581389/mnv3_resize_bilinear_test.zip)

**Comment by Saduf2019** [comment]
@ymodak 
I am able to replicate the issue on nightly, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/a9660e19ea17e5e216bcdd6ff5147745/untitled468.ipynb).Thanks!

**Comment by sunzhe09** [comment]
I met the same problem

**Comment by sunzhe09** [comment]
@Saduf2019 when I set my batch dim with 1 these layers was disappear;

**Comment by anilsathyan7** [comment]
@sunzhe09 How  did you set batch dim to 1 ? The tfllite already has batch size 1 right?

**Comment by anilsathyan7** [comment]
@Saduf2019 If we set the batch dimesion to input layer during model defintion of tf.keras model, all the problems seems to be resolved. However can we set the batch dimension of all layers for the **already trained model** to 1, before tflite conversion?

**Comment by ymodak** [comment]
The minimal example generated tflite file viewed in netron shows additonal shape, mul and strided slice layer.
<img width="381" alt="Screen Shot 2020-11-23 at 11 43 26 AM" src="https://user-images.githubusercontent.com/42785357/100007914-9e039c80-2d81-11eb-9460-524fe19835c3.png">

**Comment by karimnosseir** [comment]
This is similar to issue #43882. See my explanation on why this is correct and an example on how to set your model shape to be static before conversion

https://github.com/tensorflow/tensorflow/issues/43882#issuecomment-731636562

**Comment by Vozf** [comment]
For anyone wondering. For single input keras model the fix above can be implemented like this.
```
model.input.set_shape(1 + model.input.shape[1:])
```

**Comment by prateeks-myelin** [comment]
Referencing a similar issue i am facing with conv2dtranspose conversion to tflite after doing QAT, a reshape node is appearing which is not letting me take the graph to qualcomm ai engine direct format
https://github.com/tensorflow/model-optimization/issues/1178
